### PR TITLE
Add search filters examples for go

### PR DESCRIPTION
**Pull request by donomii**
### Type of change:

- [ ] **Documentation** updates (non-breaking change to fix/update documentation)
- [ ] **Website** updates (non-breaking change to update main page, company pages, pricing, etc)
- [ ] **Content** updates – **blog**, **podcast** (non-breaking change to add/update content)
- [ ] **Bug fix** (non-breaking change to fixes an issue with the site)
- [ ] **Feature** or **enhancements** (non-breaking change to add functionality)

### How Has This Been Tested?

- [ ] **GitHub action** – automated build completed without errors
- [ ] **Local build** - the site works as expected when running `yarn start`

> note, you can run `yarn verify-links` to test site links locally

### CODE REVIEW

File: _includes/code/howto/go/docs/mainpkg/search-filters_test.go [hunk 1]
```diff
@@ -0,0 +1,586 @@
...
+	targetID := "00037775-1432-35e5-bc59-443baaef7d80"
+	response, err := client.GraphQL().Get().
+		WithClassName("Article").
+		WithFields(graphql.Field{Name: "title"}).
+		WithWhere(filters.Where().
+			WithPath([]string{"id"}).
+			WithOperator(filters.Equal).
+			WithValueString(targetID)).
+		WithFields(
+			graphql.Field{Name: "question"},
```
Comment (daveatweaviate) [inline _includes/code/howto/go/docs/mainpkg/search-filters_test.go:405]: I think the field name should be "title"

File: _includes/code/howto/go/docs/mainpkg/search-filters_test.go [hunk 2]
```diff
@@ -0,0 +1,586 @@
...
+
+	response, err := client.GraphQL().Get().
+		WithClassName("Article").
+		WithFields(graphql.Field{Name: "title"}).
+		WithWhere(filters.Where().
+			WithPath([]string{"_creationTimeUnix"}).
+			WithOperator(filters.GreaterThan).
+			WithValueDate(timestamp)).
+		WithFields(
+			graphql.Field{Name: "question"},
```
Comment (daveatweaviate) [inline _includes/code/howto/go/docs/mainpkg/search-filters_test.go:452]: This line shouldn't be here, Articles doesn't have a question field

**Review by daveatweaviate — COMMENTED** [review]
Thanks @donomii this is a huge help

I left a couple of comments here after testing the code. Let me know if you have any questions.

There's also a comment in our slack DM about the nested filter operands. (the points can't be greater than 700 and less than 300 at the same time)

**Review by antas-marcin — APPROVED** [review]

**Review by antas-marcin — CHANGES_REQUESTED** [review]
I think we need to fix the indentation bc the Go examples are not properly indented

**Comment by daveatweaviate** [comment]
@antas-marcin  - I think GitHub wants you to approve the indent changes. Would you have a look? 

Thanks!

**Comment by antas-marcin** [comment]
@daveatweaviate the indentation is now correct, but we lost Go code syntax highlighting (look [here](https://search-filters-go--tangerine-buttercream-20c32f.netlify.app/developers/weaviate/search/filters#filter-with-multiple-conditions))

**Comment by donomii** [comment]
> @daveatweaviate the indentation is now correct, but we lost Go code syntax highlighting (look [here](https://search-filters-go--tangerine-buttercream-20c32f.netlify.app/developers/weaviate/search/filters#filter-with-multiple-conditions))

Working on this now

**Review by antas-marcin — APPROVED** [review]

**Pull request merged**